File: php/testcode/benchmark_test_00849.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00849(BenchmarkRequest $req): BenchmarkResponse {
    $wpdb = new FakeWpdb(getDbConnection());
    $id = $req->param('id');
    $result = $wpdb->query("SELECT * FROM wp_users WHERE id=$id");
    return BenchmarkResponse::ok("Rows affected: " . ($result ? '1' : '0'));
}

File: rust/testcode/benchmark_test_00849.rs
fn mongo_find(collection: &str, filter: &str) -> String {
    format!("Query on {} with filter: {}", collection, filter)
}

pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let score = req.param("score");
    let filter = format!(
        "{{\"$where\": \"this.score > {} && this.active == true\"}}",
        score
    );
    let result = mongo_find("leaderboard", &filter);
    super::shared::BenchmarkResponse::ok(&result)
}
